pkg/resources/file_format.go

```diff
@@ -0,0 +1,1018 @@
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies whether to insert SQL NULL for empty fields in an input file, which are represented by two successive delimiters.",
+	},
+	"skip_byte_order_mark": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether to skip the BOM (byte order mark), if present in a data file.",
+	},
+	"encoding": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "String (constant) that specifies the character set of the source data when loading data into a table.",
+	},
+	"enable_octal": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that enables parsing of octal numbers.",
+	},
+	"allow_duplicate": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies to allow duplicate object field names (only the last one will be preserved).",
+	},
+	"strip_outer_array": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that instructs the JSON parser to remove outer brackets.",
+	},
+	"strip_null_values": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that instructs the JSON parser to remove object fields or array elements containing null values.",
+	},
+	"ignore_utf8_errors": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether UTF-8 encoding errors produce error conditions.",
+	},
+	"snappy_compression": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether unloaded file(s) are compressed using the SNAPPY algorithm.",
+	},
+	"binary_as_text": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether to interpret columns with no defined logical data type as UTF-8 text.",
+	},
+	"preserve_space": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether the XML parser preserves leading and trailing spaces in element content.",
+	},
+	"strip_outer_element": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether the XML parser strips out the outer XML element, exposing 2nd level elements as separate documents.",
+	},
+	"disable_snowflake_data": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether the XML parser disables recognition of Snowflake semi-structured data tags.",
+	},
+	"disable_auto_convert": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Boolean that specifies whether the XML parser disables automatic conversion of numeric and Boolean values from text to native representation.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the file format.",
+	},
+}
+
+type fileFormatID struct {
```

Comment:
I refactored the way we encode and decode IDs recently in #380. You should be able to use that to drop this.

Reply:
Updated